config/locales/en.yml

```diff
@@ -0,0 +1,16 @@
+# Localization
+#
+# Replace the following placeholders with the correct values for
+# your installation.
+#
+# <NO_RESULTS_LINK>: The link for your site when there are no results
+
+en:
+  archives_space_search:
+    display_name: "The Archives"
```

Comment:
In config/locales/en.yml, I think the expectation was that the display name would be "Archives Space", not "The Archives". Did you have a particular reason for choosing that name?

Also, NCSU has accepted a pull request so that the "no_results_link" is basically not used anymore. So you can safely leave it as "<NO_RESULTS_LINK>" (see https://github.com/umd-lib/quick_search-swiftype_searcher/blob/develop/config/locales/en.yml)

Reply:
Yeah, that's what the ticket requested. I think they want to avoid "Archivesspace" because it's jargon ( nobody that doesn't work  here knows/cares what that is ). But "The Archives" sounds odd too. 

If they're not using the no_results_link should we just remove it?

Reply:
Okay -- I see where you got "The Archives" from.

There are two audiences in play here. We want the searchers to be as generic as possible, with UMD-specific configuration being done in the searchumd-special application.

The hope is that a searcher might be useful to non-UMD folks (although that is unlikely in this particular case), and so the terminology should reflect that. So I think we should definitely use "ArchivesSpace" in the searcher. If we want to change it to "The Archives" in the searchumd-special application, we can do that by simply editing the en.yml file in that application.

As for the "no_results_link" in the config/locales/en.yml file, the backstory is as follows.

In NCSU QuickSearch, there is a "loaded_link" and a "no_result_link". The "loaded_link" is intended to provide a link so that the user can go to the native search interface with the query term already filled in (so that if they want to see additional results from that search term, they just click the link). The "no_result_link" is used when no search results are found, and generally just takes the user to the native search interface page.

In NSCU QuickSearch v0.2.0 (the latest released version), the "loaded_link" is configurable either in the en.yml file, or by a "loaded_link" method in the searcher. The "no_results_link" is only configurable in the en.yml file. This is a problem when we are setting the "loaded_link" via environment variables in the .env file -- there is no way to make similar changes to the "no_results_link" value.

UMD submitted a pull request to NCSU, which they accepted, which enables a "no_results_link" property to be added to the searcher configuration file. This enables the property to be configured via the environment variables. In order to preserve backward compatibility, the QuickSearch base class still returns the "no_results_link" value (if it exists) from the en.yml file. If the "no_results_link" value exists in the en.yml file, it is always returned. The changes have not yet been released in an actual version.

The searchumd-special application is based on the NSCU QuickSearch "master" branch, so we can take advantage of this functionality. We want to use the new functionality in the "searchumd-special" application, but keep the searcher itself compatible with v0.2.0.

So we should leave the "no_results_link" in the "quick_search-archives_space_searcher/config/locales/en.yml" file.

The following changes should be made to the "searchumd-special" application:

1) add a "no_results_link" property to "searchumd-special/config/searchers/archives_space_config.yml", presumably something like:

```
defaults:
    ...
    # Link to provide when no results are found
    no_results_link: "<%= ENV['ARCHIVES_SPACE_URL'] %>"
```

Note: When I tried this, I got sent to "https://archives.lib.umd.edu/search", which displays an error. I think we want to rework things so that we have a "base url" of "https://archives.lib.umd.edu/" and a "search_url" (used for the actual searching) that includes the "/search" path. Basically when a "no results" link appears, we want to send them to the home page, not a search page that displays an error (even if they can still search from that page).

2) In the "searchumd-special" application, we should override the en.yml from the searcher by editing the "searchumd-special/config/locales/en.yml" file and adding:

``
  archives_space_search:
    no_results_link:
```
This will ensure that the value in the configuration file is used.

You can test this by running the "searchumd-special" application and putting in a nonsensical search term such as "foobarbazquuz". The search will return no results and the "Try a different search..." link should go to the ArchivesSpace home page.